Sources/ApolloCodegenLib/Templates/SelectionSetTemplate.swift

```diff
@@ -707,28 +706,116 @@ fileprivate extension IR.MergedSelections.MergedSource {
 
 fileprivate struct SelectionSetNameGenerator {
 
+  enum Format {
+    /// Fully qualifies the name of the selection set including the name of the enclosing
+    /// operation or fragment.
+    case fullyQualified
+    /// Omits the root entity selection set name
+    /// (ie. the name of the enclosing operation or fragment).
+    case omittingRoot
+  }
+
+  static func generatedSelectionSetName(
+    for selectionSet: IR.SelectionSet,
+    to toNode: LinkedList<IR.ScopeCondition>.Node? = nil,
+    format: Format,
+    pluralizer: Pluralizer
+  ) -> String {
+    generatedSelectionSetName(
+      for: selectionSet.typeInfo,
+      to: toNode,
+      format: format,
+      pluralizer: pluralizer
+    )
+  }
+
+  static func generatedSelectionSetName(
+    for mergedSource: IR.MergedSelections.MergedSource,
+    format: Format,
+    pluralizer: Pluralizer
+  ) -> String {
+    generatedSelectionSetName(
+      for: mergedSource.typeInfo,
+      format: format,
+      pluralizer: pluralizer
+    )
+  }
+
+  private static func generatedSelectionSetName(
```

Comment:
This is the core of the changes in the template. The rest is mostly updating call sites to use these new functions.

This refactors the selection set name computation to account for the fact that the `fieldPath` no longer includes a component for the root definition.